controllers/gce/controller/cluster_manager.go

```diff
@@ -155,9 +161,33 @@ func (c *ClusterManager) Checkpoint(lbs []*loadbalancers.L7RuntimeInfo, nodeName
 		np = append(np, p.Port)
 	}
 	if err := c.firewallPool.Sync(np, nodeNames); err != nil {
-		return err
+		return igs, err
 	}
 
+	return igs, nil
+}
+
+func (c *ClusterManager) CreateInstanceGroups(servicePorts []backends.ServicePort) ([]*compute.InstanceGroup, error) {
```

Comment:
Can we rename this and the other `CreateInstanceGroups`.
`EnsureInstanceGroupsAndPorts`?